### PR TITLE
feat(inventory): get tag from related agent for NetDiscovery

### PR DESCRIPTION
**Pull request by stonebuzz**
If ```TAG``` not available  (ie netDiscovery)
Try to find related ```TAG``` Agent with ```deviceID``` 

and make it available from ```Rule``` engine

| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

File: src/Inventory/Asset/MainAsset.php [hunk 1]
```diff
@@ -393,6 +394,14 @@ public function prepareAllRulesInput(\stdClass $val): array
             $input['tag'] = $this->getAgent()->fields['tag'];
         }
 
+        //form discovery mode try to get related agent TAG by it's deviceid
+        $agent = new Agent();
+        if (!isset($input['tag'])) {
```
Comment (g-bougard) [inline src/Inventory/Asset/MainAsset.php:399]:
```suggestion
        // for discovery mode, try to get TAG from agent by its deviceid
        if (!isset($input['tag'])) {
            $agent = new Agent();
```

**Review by g-bougard — CHANGES_REQUESTED** [review]
Comment is breaking my head :-P
one line can be moved into `if` block.

File: src/Inventory/Asset/MainAsset.php [hunk 2]
```diff
@@ -393,6 +394,14 @@ public function prepareAllRulesInput(\stdClass $val): array
             $input['tag'] = $this->getAgent()->fields['tag'];
         }
 
+        //form discovery mode try to get related agent TAG by it's deviceid
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply src/Inventory/Asset/MainAsset.php:397]: Done 
(it was always at ```false``` on my side, but it's a plugin bug see ->  https://github.com/glpi-project/glpi-inventory-plugin/pull/135)

File: src/Inventory/Asset/MainAsset.php (hunk 1 above)
Comment (stonebuzz) [reply src/Inventory/Asset/MainAsset.php]: XD, done

File: src/Inventory/Asset/MainAsset.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (g-bougard) [reply src/Inventory/Asset/MainAsset.php]: I guess netinventory case could be also concerned.

File: src/Inventory/Asset/MainAsset.php [hunk 3]
```diff
@@ -393,6 +394,14 @@ public function prepareAllRulesInput(\stdClass $val): array
             $input['tag'] = $this->getAgent()->fields['tag'];
         }
 
+        // for discovery mode, try to get TAG from agent by its deviceid
+        $agent = new Agent();
+        if (!isset($input['tag']) && $this->is_discovery === true ) {
```
Comment (cedric-anne) [inline src/Inventory/Asset/MainAsset.php:399]:
```suggestion
        if (!isset($input['tag']) && $this->is_discovery === true) {
```

File: src/Inventory/Asset/MainAsset.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (stonebuzz) [reply src/Inventory/Asset/MainAsset.php]: Maybe in other PR ?

**Review by trasher — APPROVED** [review]

**Pull request closed without merging**